src/components/Api.js

```diff
@@ -0,0 +1,88 @@
+export default class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  _fetch(url, options = {}) {
+    return fetch(`${this._baseUrl}${url}`, options).then((res) =>
+      res.ok ? res.json() : Promise.reject(`Error: ${res.status}`)
+    );
+  }
+
+  getUserData() {
+    return this._fetch("/users/me", {
+      method: "GET",
+      headers: this._headers,
+    });
+  }
+
+  getInitialCards() {
+    return this._fetch("/cards", {
+      method: "GET",
+      headers: this._headers,
+    });
+  }
+
+  updateUserProfile(data) {
+    return this._fetch("/users/me", {
+      method: "PATCH",
+      headers: {
+        ...this._headers,
+        "Content-Type": "application/json",
```

Comment:
According to the task you need to pass full `headers` into the constructor of `Api`. See here https://snipboard.io/pRdaAZ.jpg . And in the code you need to use only: `headers: this._headers` (not to duplicate the code)